include/boost/geometry/extensions/gis/io/wkb/detail/ogc.hpp

```diff
@@ -107,99 +107,61 @@ struct ewkt_policy
 {
 };
 
-template
-<
-    typename Geometry, 
-    typename CheckPolicy = ogc_policy, 
-    typename Tag = typename tag<Geometry>::type
->
-struct geometry_type : not_implemented<Tag>
-{
-};
-
-template <typename Geometry, typename CheckPolicy>
-struct geometry_type<Geometry, CheckPolicy, point_tag>
+template <typename Geometry,
+          geometry_type_ogc::enum_t OgcType,
```

Comment:
Ok, I addressed all of your suggestions (and more) in this commit: https://github.com/boostorg/geometry/commit/a48df2a15338cc4e38c30496f1678435c2c6faca